### DISCUSSION TITLE
How do I use remarks extensions?

### DISCUSSION THREAD

**Discussion by Gusis93** [micromark/micromark#129] [Q&A] [answered]
Hi, some of the answer to the questions are links to remarks extensions. In the readme for example you say that the answer to using breaks when the user enters is in https://github.com/remarkjs/remark-breaks. How exactly do I use this extension?

Or is it more like look at the code and write the extension for micromark?

Micromark covers everything I need except for this one thing. Thanks a lot to whoever can answer

**Comment by wooorm** [comment, marked as answer]
Use unified/remark if you want remark plugins. This project is lower level.

**Reply by ChristianMurphy** [reply]
Adding on, `remark` and `remark` plugins work on abstract syntax trees (AST).
`micromark` works on tokens (a lower level abstraction than AST).

`remark` plugins need the higher level AST abstraction to work.

You are welcome to see if you could create a token based version of the breaks extension for `micromark`, though be aware that some features are difficult to impossible to re-create without an AST, which is why `remark` exists.
